### DISCUSSION TITLE
Sample workflow json using workflow variable

### DISCUSSION THREAD

**Discussion by mremadevi** [elsa-workflows/elsa-core#5293] [Q&A]
Can I get a sample workflow json using workflow variable in branching?

**Comment by glime-ai** [comment]
Certainly! Below is a simplified example of a JSON structure for a workflow that uses a workflow variable in branching. This example assumes that you have a variable called `Decision` that can have values like "Approve" or "Reject", and based on this decision, the workflow will branch accordingly.

```json
{
  "name": "DecisionWorkflow",
  "variables": [
    {
      "name": "Decision",
      "type": "String"
    }
  ],
  "activities": [
    {
      "id": "activity1",
      "type": "Elsa.Decision",
      "properties": {
        "expression": {
          "type": "Literal",
          "value": "${getDecision()}"
        }
      },
      "outcomes": ["Approve", "Reject"]
    },
    {
      "id": "activity2",
      "type": "Elsa.Approve",
      "displayName": "Approve Branch",
      "source": "activity1",
      "sourceOutcome": "Approve"
    },
    {
      "id": "activity3",
      "type": "Elsa.Reject",
      "displayName": "Reject Branch",
      "source": "activity1",
      "sourceOutcome": "Reject"
    }
  ],
  "connections": [
    {
      "sourceId": "activity1",
      "targetId": "activity2",
      "outcome": "Approve"
    },
    {
      "sourceId": "activity1",
      "targetId": "activity3",
      "outcome": "Reject"
    }
  ]
}
```

In this JSON:

- The `variables` array defines a variable named `Decision`.
- The `activities` array contains an `Elsa.Decision` activity that evaluates the `Decision` variable.
- The `outcomes` of the `Elsa.Decision` activity are "Approve" and "Reject".
- Based on the outcome, the workflow will continue with either the `Elsa.Approve` or `Elsa.Reject` activity.
- The `connections` array defines the flow between activities based on the outcome of the decision.

Please note that the actual activity types (`Elsa.Decision`, `Elsa.Approve`, `Elsa.Reject`) and their properties will depend on the specific implementation and capabilities of the Elsa Workflow engine you are using.

For more detailed guidance and examples, you can refer to the Elsa documentation and explore the JSON representations of workflows created via the designer, as they provide insights into how to structure your workflow definitions.

This comment was generated by [Glime](https://glimelab.ai/?utm_source=dRbvTb7IhCSYiI&utm_medium=github).
